src/node/index.ts

```diff
@@ -303,7 +301,7 @@ class Node {
           } as UpdateLog),
         })
         await this.stop()
-        this.downloadAndInstall()
+        await this.downloadAndInstall()
```

Comment:
I've on purpose not added `await` keyword after the `this.downloadAndInstall()` so that updates happen in parallel when we're checking for updates. If not adding `await` does not cause any issue, then we should remove it IMO.